package com.vulnerability.dao;

import java.sql.SQLException;
import java.util.Collections;
import java.util.Map;

import javax.sql.DataSource;

import org.springframework.jdbc.core.BeanPropertyRowMapper;
import org.springframework.jdbc.core.RowMapper;
import org.springframework.jdbc.core.namedparam.BeanPropertySqlParameterSource;
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
import org.springframework.jdbc.core.namedparam.SqlParameterSource;
import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
import org.springframework.stereotype.Repository;

import com.vulnerability.dto.User;

@Repository
public class UserDao {
	
	private NamedParameterJdbcTemplate jdbc; 
	private SimpleJdbcInsert insert;
	private RowMapper<User> rowmapper = BeanPropertyRowMapper.newInstance(User.class);
	
	public UserDao(DataSource dataSource) {
		jdbc = new NamedParameterJdbcTemplate(dataSource);
		insert = new SimpleJdbcInsert(dataSource).usingGeneratedKeyColumns("id").withTableName("user");
	}
	
	
	public User getUserByLoginId(String loginId) throws SQLException{
		Map<String, Object> map = Collections.singletonMap("loginId", loginId);
		User user = jdbc.queryForObject(UserDaoSql.SELECT_BY_LOGIN_ID, map, rowmapper);
		return user;
	}
	
	public int insertUser(User user) {
		SqlParameterSource params = new BeanPropertySqlParameterSource(user);
		int num = insert.executeAndReturnKey(params).intValue();
		return num;
	}
	
	
}
